### ISSUE TITLE
JsonIgnoreProperties not working on fields

### ISSUE THREAD

**Issue by essien** [FasterXML/jackson-databind#2029]
The documentation says:  
> Starting with 2.0, this annotation can be applied both to classes and to properties.

However, I find that JsonIgnoreProperties works when applied on the class level, but fails when applied on the field. I'm using Jackson 2.8.11

The following test demonstrates this.

```java
    @JsonIgnoreProperties(value = "id", allowGetters = true)
    private static class TestClass1 {

        private String id;

        public String getId() {
            return id;
        }

        public void setId(String id) {
            this.id = id;
        }
    }

    private static class TestClass2 {

        @JsonIgnoreProperties(value = "id", allowGetters = true)
        private String id;

        public String getId() {
            return id;
        }

        public void setId(String id) {
            this.id = id;
        }
    }

    @Test // This test case passes
    public void testIgnorePropertiesOnClass() throws Exception {
        ObjectMapper objectMapper = new ObjectMapper();
        TestClass1 testClass = new TestClass1();
        testClass.setId("abcd");
        String writeValueAsString = objectMapper.writeValueAsString(testClass);
        Assertions.assertThat(writeValueAsString).contains("abcd");
        
        TestClass1 readValue = objectMapper.readValue("{\"id\": \"myid\"}", TestClass1.class);
        Assertions.assertThat(readValue.getId()).isNull();
    }

    @Test // This test case fails
    public void testIgnorePropertiesOnField() throws Exception {
        ObjectMapper objectMapper = new ObjectMapper();
        TestClass2 testClass = new TestClass2();
        testClass.setId("abcd");
        String writeValueAsString = objectMapper.writeValueAsString(testClass);
        Assertions.assertThat(writeValueAsString).contains("abcd");
        
        TestClass2 readValue = objectMapper.readValue("{\"id\": \"myid\"}", TestClass2.class);
        Assertions.assertThat(readValue.getId()).isNull(); // This assertion fails.
    }
```

**Comment by cowtowncoder** [comment]
This might be misunderstanding on the meaning of what adding `@JsonIgnoreProperties` on the field means: it applies to POJO value, so you could use it like:

```java
public class Stuff {
  @JsonIgnoreProperties(value = "id", allowGetters = true)
  public TestClass1 value;
}
```

and then `value` would not include `id`.

In your case I think what you are looking for is instead

```java
    private static class TestClass2 {
      @JsonProperty(access = JsonProperty.Access.READ_ONLY)
        private String id;
   }
```

I hope this helps!

**Comment by essien** [comment]
You're right, `@JsonProperty(access = JsonProperty.Access.READ_ONLY)` was what I was looking for. In fact I've been using it with Jackson 2.8.x and it's been working for me. But ever since I upgraded to Jackson 2.9.0 and above, I noticed that Swagger stopped understanding that fields marked with `READ_ONLY` should be hidden in the model for `POST` requests. I was going to open an issue for this.

But just before that, I wanted to find alternate ways of marking a field as read-only and `JsonProperties` came to the rescue. Swagger understood it and did what I wanted. But the challenge is that `JsonProperties` when applied on a class, overrides the `JsonProperties` annotation in the super class. So I decided to try applying it on the field so that I can benefit from inheritance.

However, from your explanation now it's clear that `@JsonProperties` cannot be applied in this scenario. I guess I'm stuck and have to manually copy the `JsonProperties.value` in the super class and append them to the `JsonProperties.value` in the subclass. Would have been nice to have a way to automatically copy/merge them to the subclass. I'll appreciate any recommendation.
